### PR TITLE
Add tree import and export

### PR DESCRIPTION
**Pull request by jakub-drak-sbahi**
image

image

### CODE REVIEW

File: web/src/app/app.module.ts [hunk 1]
```diff
@@ -48,7 +50,9 @@ import { PosterComponent } from './import-export/poster/poster.component';
     PersonFilesFormComponent,
     TreeDrawComponent,
     LicensesComponent,
-    PosterComponent
+    PosterComponent,
+    ImportComponent,
```
Comment (BartoszChrostowski) [inline web/src/app/app.module.ts:54]: Does this add import too?

File: web/src/app/import-export/export/export.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  constructor(
+    private service: ImportExportService
+  ) { }
+
+  ngOnInit(): void {
+  }
+
+  downloadFile() {
+    this.service.getTreeFileExport(this.treeId).subscribe(result => {
+      var blob = new Blob([result], { type: 'text/xml' });
```
Comment (BartoszChrostowski) [inline web/src/app/import-export/export/export.component.ts:22]: Can you add mapping from file extension? There it is not that important, but probably when adding persons files you should use some library that maps file extension to mime type, forgot to check that in "Add persons file" Merge Request

File: web/src/app/import-export/export/export.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    var url = window.URL.createObjectURL(blob);
+    var filename = `tree_${this.treeId}.xml`;
+
+    var a = document.createElement("a");
+    // @ts-ignore
+    a.style = "display: none";
+    a.href = url;
+    a.download = filename;
+
+    // IE 11
```
Comment (BartoszChrostowski) [inline web/src/app/import-export/export/export.component.ts:37]: We support IE 11? 😍

**Review by BartoszChrostowski — APPROVED** [review]

File: web/src/app/import-export/export/export.component.ts (hunk 3 above)
Comment (jakub-drak-sbahi) [reply web/src/app/import-export/export/export.component.ts:37]: With downloading files - yes 😄

File: web/src/app/app.module.ts (hunk 1 above)
Comment (jakub-drak-sbahi) [reply web/src/app/app.module.ts:54]: Yes

File: web/src/app/import-export/export/export.component.ts (hunk 2 above)
Comment (jakub-drak-sbahi) [reply web/src/app/import-export/export/export.component.ts]: It is export not import, but now I changed it and I read type of file from response.

**Pull request merged**